target/linux/ipq40xx/patches-5.4/200-arm-compressed-add-appended-DTB-section.patch

```diff
@@ -0,0 +1,49 @@
+From 428baaf5091d41170ca3f556a629c6c8b8c7985c Mon Sep 17 00:00:00 2001
+From: Robert Marko <robimarko@gmail.com>
+Date: Fri, 30 Oct 2020 13:36:31 +0100
+Subject: [PATCH] arm: compressed: add appended DTB section
+
+This adds a appended_dtb section to the ARM decompressor
+linker script.
+
+This allows using the existing ARM zImage appended DTB support for
+appending a DTB to the raw ELF kernel.
+
+Its size is set to 1MB max to match the zImage appended DTB size limit.
+
+To use it to pass the DTB to the kernel, objcopy is used:
+
+objcopy --set-section-flags=.appended_dtb=alloc,contents \
+	--update-section=.appended_dtb=<target>.dtb vmlinux
+
+This is based off the following patch:
+https://github.com/openwrt/openwrt/commit/c063e27e02a9dcac0e7f5877fb154e58fa3e1a69
+
+Signed-off-by: Robert Marko <robimarko@gmail.com>
+---
+ arch/arm/boot/compressed/vmlinux.lds.S | 10 +++++++++-
+ 1 file changed, 9 insertions(+), 1 deletion(-)
+
+--- a/arch/arm/boot/compressed/vmlinux.lds.S
++++ b/arch/arm/boot/compressed/vmlinux.lds.S
+@@ -93,6 +93,14 @@ SECTIONS
+ 
+   _edata = .;
+ 
++  .appended_dtb : {
++    __appended_dtb = .;
```

Comment:
This __appended_dtb symbol exists because it was referenced:
https://elixir.bootlin.com/linux/latest/source/arch/mips/boot/compressed/decompress.c#L36

It can be dropped for arm.

Reply:
Thanks for catching it.
Dropped it and pushed a update.